master_server/master_server.py

```diff
@@ -0,0 +1,221 @@
+    This class configures the flask app and executes function
+    pre_task before starting flask server.
+  """
+  def run(self, host=None, port=None, debug=None, load_dotenv=True, **options):
+    if not self.debug or os.getenv('WERKZEUG_RUN_MAIN') == 'true':
+      with self.app_context():
+        pre_tasks()
+    super(MyFlaskApp, self).run(
+        host=host, port=port, debug=debug, load_dotenv=load_dotenv, **options)
+
+app = MyFlaskApp(__name__)
+
+app.config['SQLALCHEMY_DATABASE_URI'] = 'sqlite:///database.db'
+db = SQLAlchemy(app)
+
+
+working_vm_address = []
+available_vm_address = []
+
+class Newrequestentry(db.Model):
+  """
+    This class creates a table Newrequestentry
+    init function in this class inserts a new row
+    the table.
+  """
+  request_id = db.Column(db.Integer, primary_key=True, autoincrement=True)
+  proto_file = db.Column(db.BLOB)
+  request_status = db.Column(db.String(50))
+  last_vm_assigned = db.Column(db.String(100))
+  number_of_retries_allowed = db.Column(db.Integer)
+  number_of_retries_till_now = db.Column(db.Integer)
+
+  def __init__(
+      self, proto_file, request_status, last_vm_assigned,
+      number_of_retries_allowed, number_of_retries_till_now):
+    self.proto_file = proto_file
+    self.request_status = request_status
+    self.last_vm_assigned = last_vm_assigned
+    self.number_of_retries_allowed = number_of_retries_allowed
+    self.number_of_retries_till_now = number_of_retries_till_now
+
+def create_new_request(
+    proto_file, request_status, last_vm_assigned,
+    number_of_retries_allowed, number_of_retries_till_now):
+  """This function adds a new row to database.
+  """
+  dessert = Newrequestentry(
+      proto_file, request_status, last_vm_assigned, number_of_retries_allowed,
+      number_of_retries_till_now)
+  db.session.add(dessert)
+  db.session.commit()
+  return dessert
+
+def get_status(request_id):
+  """Get status of previous request.
+  """
+  get_row_data = Newrequestentry.query.get(request_id)
+  return get_row_data
+
+def change_state(request_id):
+  """Change request_status of a request.
+  """
+  Newrequestentry.query.filter_by(request_id=request_id).\
+  update({Newrequestentry.request_status: 'completed'})
+  db.session.commit()
+
+@app.route('/', methods=['GET', 'POST'])
+def hello_world():
+  print(' You have reached master server \n')
+
+@app.route('/register', methods=['GET', 'POST'])
+def add_vm_address():
+  """Add a VM to working_vm_address list.
+  """
+  address = request.data.decode('utf-8')
+  if address not in working_vm_address:
+    working_vm_address.append(address)
+  return {'message': 'success'}
+
+@app.route('/upload')
+def upload_file():
+  """Serve web page
+  """
+  return render_template('upload.html')
+
+@app.route('/uploader', methods=['GET', 'POST'])
+def upload_fil():
+  """
+    Accept proto file from user and
+    assign it to VM.
+  """
+  f = request.files['file']
+  read = f.read()
+  vm_n = find_working_vm()
+  print(vm_n)
```

Comment:
Let's avoid print, you can use logger and log these at debug level

Reply:
Removed print statement.